crates/ruff_cli/src/lib.rs

```diff
@@ -85,7 +85,11 @@ fn change_detected(paths: &[PathBuf]) -> Option<ChangeKind> {
 fn is_stdin(files: &[PathBuf], stdin_filename: Option<&Path>) -> bool {
     // If the user provided a `--stdin-filename`, always read from standard input.
     if stdin_filename.is_some() {
-        if let Some(file) = files.iter().find(|file| file.as_path() != Path::new("-")) {
+        if let Some(file) = files
+            .iter()
+            // We allow `.` in addition to `-` since it is the default file selector
+            .find(|file| file.as_path() != Path::new("-") && file.as_path() != Path::new("."))
```

Comment:
Does this mean we now allow `cat foo.py | ruff check --stdin-filename="foo.py" .`, without warning? That feels like a bug. I'm wondering if we should instead represent this default as the empty slice? I.e., at the relevant sites, if `files` is empty, use the current working directory?

Reply:
Yeah... I brought this up at https://github.com/astral-sh/ruff/pull/8791#discussion_r1399516759

I don't like either solution honestly. `.` as the Clap default is nice for the CLI help menu and such and it seems like a hassle to make sure the behavior is correct in the call sites. It does feel bad to omit the warning though so I guess I'll do the extra work.

Reply:
Ahh sorry, I missed your comment! I defer to you as the author, I see there are tradeoffs here.

Reply:
Okay kind of awkward but fine https://github.com/astral-sh/ruff/pull/8791/commits/e92000d9064c9c3638124ea94d1091065c8e9e3e